mcuboot: BOOT_MAX_ALIGN is redefined

**Describe the bug**

`BOOT_MAX_ALIGN` is redefined in `include/zephyr/dfu/mcuboot.h` (already defined in `bootutil_public.h` in the module).

**To Reproduce**

```
scripts/twister -v -N -M -p nucleo_h753zi -s samples/subsys/usb/dfu/sample.usb.dfu
```

**Expected behavior**

`BOOT_MAX_ALIGN` is only defined once.

**Impact**

Unable to build tests and samples using mcuboot.

**Logs and console output**

```
-- Application: /__w/zephyr/zephyr/samples/subsys/usb/dfu
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Cache files will be written to: /__w/zephyr/zephyr/.cache
-- Zephyr version: 3.2.0-rc2 (/__w/zephyr/zephyr)
-- Found west (found suitable version "0.13.1", minimum required is "0.7.1")
-- Board: nucleo_h753zi
-- Found host-tools: zephyr 0.15.0 (/opt/toolchains/zephyr-sdk-0.15.0)
-- Found toolchain: zephyr 0.15.0 (/opt/toolchains/zephyr-sdk-0.15.0)
-- Found Dtc: /opt/toolchains/zephyr-sdk-0.15.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /__w/zephyr/zephyr/boards/arm/nucleo_h753zi/nucleo_h753zi.dts
-- Generated zephyr.dts: /__w/zephyr/zephyr/twister-out/nucleo_h753zi/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /__w/zephyr/zephyr/twister-out/nucleo_h753zi/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /__w/zephyr/zephyr/twister-out/nucleo_h753zi/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/dts.cmake
/__w/zephyr/zephyr/twister-out/nucleo_h753zi/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/zephyr.dts:490.7-519.5: Warning (simple_bus_reg): /soc/can: missing or empty reg/ranges property
Parsing /__w/zephyr/zephyr/samples/subsys/usb/dfu/Kconfig
Loaded configuration '/__w/zephyr/zephyr/boards/arm/nucleo_h753zi/nucleo_h753zi_defconfig'
Merged configuration '/__w/zephyr/zephyr/samples/subsys/usb/dfu/prj.conf'
Configuration saved to '/__w/zephyr/zephyr/twister-out/nucleo_h753zi/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/.config'
Kconfig header saved to '/__w/zephyr/zephyr/twister-out/nucleo_h753zi/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
CMake Warning at ../../../../subsys/usb/device/CMakeLists.txt:22 (message):
  CONFIG_USB_DEVICE_VID has default value 0x2FE3.

  This value is only for testing and MUST be configured for USB products.


-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    TC_RUNID


-- Build files have been written to: /__w/zephyr/zephyr/twister-out/nucleo_h753zi/samples/subsys/usb/dfu/sample.usb.dfu
[1/191] Preparing syscall dependency handling

[2/191] Generating include/generated/version.h
-- Zephyr version: 3.2.0-rc2 (/__w/zephyr/zephyr), build: v3.2.0-rc2
[3/191] Generating misc/generated/syscalls_subdirs.trigger
[4/191] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[5/191] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[6/191] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
[7/191] Generating include/generated/driver-validation.h
[8/191] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
[9/191] Generating include/generated/offsets.h
[10/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32c_sw.c.obj
[11/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32_sw.c.obj
[12/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc7_sw.c.obj
[13/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/hex.c.obj
[14/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc8_sw.c.obj
[15/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/dec.c.obj
[16/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc16_sw.c.obj
[17/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf.c.obj
[18/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/multi_heap.c.obj
[19/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/timeutil.c.obj
[20/191] Building C object zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj
[21/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj
[22/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj
[23/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/notify.c.obj
[24/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj
[25/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj
[26/191] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32h7/mpu_regions.c.obj
[27/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj
[28/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj
[29/191] Building C object CMakeFiles/app.dir/src/main.c.obj
[30/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj
[31/191] Linking C static library app/libapp.a
[32/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap-validate.c.obj
[33/191] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/common/stm32cube_hal.c.obj
[34/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/bitarray.c.obj
[35/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/onoff.c.obj
[36/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/rb.c.obj
[37/191] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32h7/soc_m7.c.obj
[38/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap.c.obj
[39/191] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/mpsc_pbuf.c.obj
[40/191] Generating linker_zephyr_pre0.cmd
[41/191] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/common/soc_config.c.obj
[42/191] Generating linker_zephyr_pre1.cmd
[43/191] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/cpu_idle.S.obj
[44/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_mgmt.c.obj
[45/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_core.c.obj
[46/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/dfu/img_util/flash_img.c.obj
[47/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_msg.c.obj
[48/191] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi_on_reset.S.obj
[49/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_backend_uart.c.obj
[50/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_output.c.obj
[51/191] Building C object zephyr/arch/common/CMakeFiles/arch__common.dir/sw_isr_common.c.obj
[52/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj
[53/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/storage/flash_map/flash_map_layout.c.obj
[54/191] Building C object zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj
[55/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/storage/flash_map/flash_map_default.c.obj
[56/191] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj
[57/191] Linking C static library zephyr/arch/common/libisr_tables.a
[58/191] Linking C static library zephyr/arch/common/libarch__common.a
[59/191] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fault_s.S.obj
[60/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/storage/flash_map/flash_map.c.obj
[61/191] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/isr_wrapper.S.obj
[62/191] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/exc_exit.S.obj
[63/191] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/reset.S.obj
[64/191] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/vector_table.S.obj
[65/191] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/irq_manage.c.obj
[66/191] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/perror.c.obj
[67/191] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi.c.obj
[68/191] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/fatal.c.obj
[69/191] Generating ../../../include/generated/libc/minimal/strerror_table.h
[70/191] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/prep_c.c.obj
[71/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/storage/stream/stream_flash.c.obj
[72/191] Building C object zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj
[73/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/atoi.c.obj
[74/191] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap.c.obj
[75/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtol.c.obj
[76/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/usb/device/usb_transfer.c.obj
[77/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtoul.c.obj
[78/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtoll.c.obj
[79/191] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/irq_init.c.obj
[80/191] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fpu.c.obj
[81/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtoull.c.obj
[82/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/bsearch.c.obj
[83/191] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/thread.c.obj
[84/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strerror.c.obj
[85/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strncasecmp.c.obj
[86/191] Linking C static library zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a
[87/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strstr.c.obj
[88/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/qsort.c.obj
[89/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strspn.c.obj
[90/191] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/thread_abort.c.obj
[91/191] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/scb.c.obj
[92/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/math/sqrtf.c.obj
[93/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/sprintf.c.obj
[94/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/fprintf.c.obj
[95/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/math/sqrt.c.obj
[96/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/string.c.obj
[97/191] Building C object zephyr/arch/arch/arm/core/aarch32/mpu/CMakeFiles/arch__arm__core__aarch32__mpu.dir/arm_core_mpu.c.obj
[98/191] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/pthread_common.c.obj
[99/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/abort.c.obj
[100/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/usb/device/usb_descriptor.c.obj
[101/191] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/nanosleep.c.obj
[102/191] Linking C static library zephyr/lib/posix/liblib__posix.a
[103/191] Building C object zephyr/arch/arch/arm/core/aarch32/mpu/CMakeFiles/arch__arm__core__aarch32__mpu.dir/arm_mpu.c.obj
[104/191] Linking C static library zephyr/arch/arch/arm/core/aarch32/mpu/libarch__arm__core__aarch32__mpu.a
[105/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/exit.c.obj
[106/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/malloc.c.obj
[107/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/stdout_console.c.obj
[108/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/usb/device/class/dfu/usb_dfu.c.obj
[109/191] Building C object zephyr/subsys/dfu/boot/CMakeFiles/subsys__dfu__boot.dir/mcuboot.c.obj
FAILED: zephyr/subsys/dfu/boot/CMakeFiles/subsys__dfu__boot.dir/mcuboot.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCORE_CM7 -DHSE_VALUE=8000000 -DKERNEL -DSTM32H753xx -DUSE_FULL_LL_DRIVER -DUSE_HAL_DRIVER -D_FORTIFY_SOURCE=2 -D__PROGRAM_START -D__ZEPHYR__=1 -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/arm/st_stm32/stm32h7 -I../../../../../../../drivers -I../../../../../../../soc/arm/st_stm32/common -I../../../../../../../subsys/usb/device -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I../../../../../../../modules/hal_nxp/usb/. -I../../../../../../../modules/hal_nxp/. -I/__w/zephyr/modules/hal/stm32/stm32cube/stm32h7xx/soc -I/__w/zephyr/modules/hal/stm32/stm32cube/stm32h7xx/drivers/include -I/__w/zephyr/modules/hal/stm32/stm32cube/stm32h7xx/drivers/include/Legacy -I/__w/zephyr/modules/hal/stm32/stm32cube/common_ll/include -I/__w/zephyr/bootloader/mcuboot/boot/bootutil/zephyr/.. -I/__w/zephyr/bootloader/mcuboot/boot/bootutil/zephyr/../include -I/__w/zephyr/bootloader/mcuboot/boot/bootutil/zephyr/../../zephyr/include -isystem ../../../../../../../lib/libc/minimal/include -isystem /opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include -isystem /opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include-fixed -fno-strict-aliasing -Os -imacros /__w/zephyr/zephyr/twister-out/nucleo_h753zi/samples/subsys/usb/dfu/sample.usb.dfu/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=/opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/arm-zephyr-eabi -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -Werror -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/samples/subsys/usb/dfu=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -std=c99 -nostdinc -MD -MT zephyr/subsys/dfu/boot/CMakeFiles/subsys__dfu__boot.dir/mcuboot.c.obj -MF zephyr/subsys/dfu/boot/CMakeFiles/subsys__dfu__boot.dir/mcuboot.c.obj.d -o zephyr/subsys/dfu/boot/CMakeFiles/subsys__dfu__boot.dir/mcuboot.c.obj -c /__w/zephyr/zephyr/subsys/dfu/boot/mcuboot.c
In file included from /__w/zephyr/zephyr/subsys/dfu/boot/mcuboot.c:20:
../../../../../../../include/zephyr/dfu/mcuboot.h:68: error: "BOOT_MAX_ALIGN" redefined [-Werror]
   68 | #define BOOT_MAX_ALIGN          8
      | 
In file included from /__w/zephyr/zephyr/subsys/dfu/boot/mcuboot.c:19:
/__w/zephyr/bootloader/mcuboot/boot/bootutil/zephyr/../include/bootutil/bootutil_public.h:89: note: this is the location of the previous definition
   89 | #define BOOT_MAX_ALIGN          MCUBOOT_BOOT_MAX_ALIGN
      | 
cc1: all warnings being treated as errors
[110/191] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/time/gmtime.c.obj
[111/191] Building C object zephyr/drivers/console/CMakeFiles/drivers__console.dir/uart_console.c.obj
[112/191] Building C object zephyr/drivers/interrupt_controller/CMakeFiles/drivers__interrupt_controller.dir/intc_exti_stm32.c.obj
[113/191] Building C object zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_page_layout.c.obj
[114/191] Building C object zephyr/drivers/hwinfo/CMakeFiles/drivers__hwinfo.dir/hwinfo_weak_impl.c.obj
[115/191] Building C object zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_stm32_ll_h7.c.obj
[116/191] Building C object zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_stm32.c.obj
[117/191] Building C object zephyr/drivers/hwinfo/CMakeFiles/drivers__hwinfo.dir/hwinfo_stm32.c.obj
[118/191] Building C object zephyr/drivers/timer/CMakeFiles/drivers__timer.dir/sys_clock_init.c.obj
[119/191] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fault.c.obj
[120/191] Building C object zephyr/drivers/timer/CMakeFiles/drivers__timer.dir/cortex_m_systick.c.obj
[121/191] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/common.c.obj
[122/191] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_stm32.c.obj
[123/191] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/usb/device/usb_device.c.obj
[124/191] Building C object zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_stm32.c.obj
[125/191] Building C object zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_stm32h7x.c.obj
[126/191] Building C object zephyr/drivers/usb/device/CMakeFiles/drivers__usb__device.dir/usb_dc_stm32.c.obj
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Toolchain: Zephyr SDK 0.15.0
 - Commit SHA: 2829de75d1ae88dc53bc6e85d8f8e59ac3cca8d3

--- a/include/zephyr/dfu/mcuboot.h
+++ b/include/zephyr/dfu/mcuboot.h
@@ -64,15 +64,6 @@ extern "C" {
 
 #define BOOT_IMG_VER_STRLEN_MAX 25  /* 255.255.65535.4294967295\0 */
 
-/* Trailer: */
-#ifndef BOOT_MAX_ALIGN
-#define BOOT_MAX_ALIGN		8
-#endif
-
-#ifndef BOOT_MAGIC_SZ
-#define BOOT_MAGIC_SZ		16
-#endif
-
 #define BOOT_TRAILER_IMG_STATUS_OFFS(bank_area) ((bank_area)->fa_size -\
 						  BOOT_MAGIC_SZ -\
 						  BOOT_MAX_ALIGN * 2)

--- a/include/zephyr/dfu/mcuboot.h
+++ b/include/zephyr/dfu/mcuboot.h
@@ -65,7 +65,10 @@ extern "C" {
 #define BOOT_IMG_VER_STRLEN_MAX 25  /* 255.255.65535.4294967295\0 */
 
 /* Trailer: */
+#ifndef BOOT_MAX_ALIGN
 #define BOOT_MAX_ALIGN		8
+#endif
+
 #ifndef BOOT_MAGIC_SZ
 #define BOOT_MAGIC_SZ		16
 #endif

--- a/subsys/dfu/img_util/CMakeLists.txt
+++ b/subsys/dfu/img_util/CMakeLists.txt
@@ -1,3 +1,5 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_sources_ifdef(CONFIG_MCUBOOT_IMG_MANAGER flash_img.c)
+
+zephyr_library_link_libraries(MCUBOOT_BOOTUTIL)

--- a/subsys/dfu/img_util/flash_img.c
+++ b/subsys/dfu/img_util/flash_img.c
@@ -14,6 +14,7 @@
 #include <zephyr/storage/stream_flash.h>
 
 #ifdef CONFIG_IMG_ERASE_PROGRESSIVELY
+#include <bootutil/bootutil_public.h>
 #include <zephyr/dfu/mcuboot.h>
 #endif
 

--- a/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/CMakeLists.txt
+++ b/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/CMakeLists.txt
@@ -15,3 +15,7 @@ zephyr_library_sources(
     src/img_mgmt_state.c
     src/img_mgmt_util.c
 )
+
+if(CONFIG_MCUBOOT_IMG_MANAGER)
+    zephyr_library_link_libraries(MCUBOOT_BOOTUTIL)
+endif()

--- a/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/zephyr_img_mgmt.c
+++ b/subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/zephyr_img_mgmt.c
@@ -13,6 +13,7 @@ LOG_MODULE_REGISTER(mcumgr_img_mgmt, CONFIG_MCUMGR_IMG_MGMT_LOG_LEVEL);
 #include <zephyr/kernel.h>
 #include <soc.h>
 #include <zephyr/init.h>
+#include <bootutil/bootutil_public.h>
 #include <zephyr/dfu/mcuboot.h>
 #include <zephyr/dfu/flash_img.h>
 #include <zephyr/mgmt/mcumgr/buf.h>
@@ -456,6 +457,12 @@ img_mgmt_impl_erase_image_data(unsigned int off, unsigned int num_bytes)
 
 	LOG_INF("Erased 0x%zx bytes of image slot", erase_size);
 
+#ifdef CONFIG_MCUBOOT_IMG_MANAGER
+	/* Right now MCUmgr supports only mcuboot images.
+	 * Above compilation swich might help to recognize mcuboot related
+	 * code when supports for anothe bootloader will be introduced.
+	 */
+
 	/* erase the image trailer area if it was not erased */
 	off = BOOT_TRAILER_IMG_STATUS_OFFS(fa);
 	if (off >= erase_size) {
@@ -474,7 +481,7 @@ img_mgmt_impl_erase_image_data(unsigned int off, unsigned int num_bytes)
 
 		LOG_INF("Erased 0x%zx bytes of image slot trailer", erase_size);
 	}
-
+#endif
 	rc = 0;
 
 end_fa:

--- a/tests/subsys/dfu/mcuboot/CMakeLists.txt
+++ b/tests/subsys/dfu/mcuboot/CMakeLists.txt
@@ -6,3 +6,5 @@ project(mcuboot)
 
 FILE(GLOB app_sources src/*.c)
 target_sources(app PRIVATE ${app_sources})
+
+zephyr_library_link_libraries(MCUBOOT_BOOTUTIL)

--- a/tests/subsys/dfu/mcuboot/src/main.c
+++ b/tests/subsys/dfu/mcuboot/src/main.c
@@ -6,6 +6,7 @@
 
 #include <zephyr/ztest.h>
 #include <zephyr/storage/flash_map.h>
+#include <bootutil/bootutil_public.h>
 #include <zephyr/dfu/mcuboot.h>
 
 #define SLOT0_PARTITION		slot0_partition

--- a/tests/subsys/dfu/mcuboot_multi/CMakeLists.txt
+++ b/tests/subsys/dfu/mcuboot_multi/CMakeLists.txt
@@ -6,3 +6,5 @@ project(mcuboot)
 
 FILE(GLOB app_sources src/*.c)
 target_sources(app PRIVATE ${app_sources})
+
+zephyr_library_link_libraries(MCUBOOT_BOOTUTIL)

--- a/tests/subsys/dfu/mcuboot_multi/src/main.c
+++ b/tests/subsys/dfu/mcuboot_multi/src/main.c
@@ -6,6 +6,7 @@
 
 #include <zephyr/ztest.h>
 #include <zephyr/storage/flash_map.h>
+#include <bootutil/bootutil_public.h>
 #include <zephyr/dfu/mcuboot.h>
 #include <zephyr/drivers/flash.h>
 

